twill-api/src/main/java/org/apache/twill/api/TwillController.java

```diff
@@ -89,4 +90,30 @@
    * @return A {@link Future} that will be completed when the restart operation has been done.
    */
   Future<String> restartInstances(String runnable, int instanceId, int... moreInstanceIds);
+
+  /**
+   * Set the log levels for requested logger names for Twill applications running in a container.
+   * The log level for a logger name can be {@code null} except for the root logger, which will reset the log level for
```

Comment:
We check it using `Preconditions.checkArgument`, so an exception will be thrown for null log level for root logger.